Data/OldList.hs

```diff
@@ -249,8 +249,16 @@ stripPrefix _ _ = Nothing
 -- | The 'elemIndex' function returns the index of the first element
 -- in the given list which is equal (by '==') to the query element,
 -- or 'Nothing' if there is no such element.
-elemIndex       :: Eq a => a -> [a] -> Maybe Int
-elemIndex x     = findIndex (x==)
+-- elemIndex x = findIndex (x==) 
+elemIndex x xs  = let elemIndex' x xs = findIndex (x==) xs
+                      !x' = x
+                      !x_as_list = [x']
```

Comment:
We should define these in the where block at line 258.  The reason is to avoid a subtle problem with strictness if we are not using smt-strings and call elemIndex- in particular, the following is from GHCi:
```
ghci> elemIndex undefined []
Nothing
```
If the list is empty, the element can be undefined, and the function will still return Nothing- but this won't work if the element is always strictly evaluated.